atlp-rwanda/e-comm-team-brogrammers-bn
# #184581327  A Seller Should see statistics in a given time-frame

#### What does this PR do?

- To let  A Seller see their statistics in a given time-frame

#### Description of Task to be completed?

display key metrics such as: 
1. total revenue
2. number of orders
3. products sold
4. money made in a time frame
5. and top-selling products

#### How should this be manually tested?

- in terminal add `npm i`
- add `.env` in your `repo` and assign the variables as there are in `.env.example` file
- run in terminal `npm run migrate:all` to setup the database
- run in terminal `npm start` to start the project
- in your browser add `http://localhost:<your port>/api-docs` for documentation
- Navigate to route `users/stats` to see your statistics as a `seller`

#### Any background context you want to provide?

A Seller statistics

#### What are the relevant pivotal tracker/Trello stories?

https://www.pivotaltracker.com/story/show/184581327

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2023-04-03 at 09 41 56" src="https://user-images.githubusercontent.com/118928495/229448302-3eea2481-8850-45cc-8f8e-548c77a437e4.png">
<img width="1440" alt="Screen Shot 2023-04-03 at 09 05 15" src="https://user-images.githubusercontent.com/118928495/229448365-4b460596-d95c-4cf2-b84d-9f622815d975.png">
<img width="1440" alt="Screen Shot 2023-04-03 at 10 04 58" src="https://user-images.githubusercontent.com/118928495/229449250-d3bbc7aa-0c80-47ea-a17c-2ac86953894c.png">
<img width="1440" alt="Screen Shot 2023-04-03 at 10 05 06" src="https://user-images.githubusercontent.com/118928495/229449278-2a1a9942-876b-4242-90d7-6512e742aea1.png">
<img width="1440" alt="Screen Shot 2023-04-03 at 10 05 38" src="https://user-images.githubusercontent.com/118928495/229449334-2b25fbe4-955f-4e58-a5c1-6c3766f2565e.png">
<img width="1440" alt="Screen Shot 2023-04-03 at 10 07 13" src="https://user-images.githubusercontent.com/118928495/229449483-875cd9a7-c413-4bac-b2c7-a6a3a85eb72e.png">
<img width="1440" alt="Screen Shot 2023-04-04 at 16 12 56" src="https://user-images.githubusercontent.com/112763782/229820653-7ad98888-53cf-446a-a1c8-0e6bae57f7bb.png">


#### Questions:

n/a
